### PR TITLE
Changed gradient_clip_val, and gradient_clip_algorithm

### LINKED ISSUE #571: Batch size is confusing in multi-GPU settings
When using `n` gpus, the actual batch size passed to each GPU is `batch_size/n`, not `batch_size`. This can be confusing and I think a small remark in the config file and the documentation could help.

### PR DESCRIPTION
**Pull request by bhavyak7-lab**
Addresses #571. The dim_intensity value in the config.yaml file is not used anywhere in the code, so it can be removed.

## Summary by CodeRabbit

* **Configuration**
  * Gradient clipping set to norm with value 1.0; clarified per-device micro-batch behavior for multi-GPU training.

* **New Features**
  * Added support for timsTOF .d-folder spectra input.

* **Documentation**
  * Updated supported MS/MS file formats and changelog; minimum Python requirement raised to 3.10.

* **Tests**
  * Tests now validate peptide masses numerically, use parameterized enzyme cases, expect formatted scan IDs, verify tokenizer compatibility, and adjust integration expectations.

* **Refactor / Style**
  * Removed some module-level logger initializations and minor formatting cleanups.

### CODE REVIEW

**Comment by bittremieux** [comment]
@bhavyak7-lab please rebase to `dev` instead of `main`.

I also recommend to keep the original description for the batch size, and then add a clarification when using multiple GPUs on a subsequent line.

File: casanovo/denovo/model_runner.py [hunk 1]
```diff
@@ -525,12 +523,18 @@ def initialize_model(self, train: bool, db_search: bool = False) -> None:
...
-                    warnings.warn(
-                        f"Mismatching {param} parameter in "
-                        f"model checkpoint ({self.model.hparams[param]}) "
-                        f"vs config file ({model_params[param]}); "
-                        "using the checkpoint."
-                    )
+                    if param == "tokenizer":
+                        self._verify_tokenizer(
+                            model_params[param], self.model.hparams[param]
+                        )
```
[the remark this answers is not in the document]
Comment (bittremieux) [reply casanovo/denovo/model_runner.py:529]: **todo:** This comment seems correct. Why did the order of these parameters need to be changed?

File: casanovo/denovo/model_runner.py [hunk 2]
```diff
@@ -523,9 +525,9 @@ def initialize_model(self, train: bool, db_search: bool = False) -> None:
                 if model_params[param] != self.model.hparams[param]:
                     if param == "tokenizer":
                         self._verify_tokenizer(
-                            model_params[param], self.model.hparams[param]
+                            self.model.hparams[param], model_params[param]
                         )
-                    else:
+
```
Comment (bittremieux) [inline casanovo/denovo/model_runner.py:530]: **todo:** I think the `else` was correct?

File: casanovo/denovo/model_runner.py [hunk 3]
```diff
@@ -622,7 +624,7 @@ def _verify_tokenizer(
 
         logger.warning(
             "Mismatching %s in model checkpoint tokenizer vs config file tokenizer;"
-            " using the checkpoint tokenizer.",
+            " using the config tokenizer.",
```
Comment (bittremieux) [inline casanovo/denovo/model_runner.py:627]: **question:** Is the config tokenizer actually used?

File: casanovo/denovo/model_runner.py [hunk 4]
```diff
@@ -652,7 +654,7 @@ def _get_input_paths(
         """
         ext = (".mgf", ".lance")
         if not annotated:
-            ext += (".mzml", ".mzxml")
+            ext += (".mzml", ".mzxml", ".d")
```
Comment (bittremieux) [inline casanovo/denovo/model_runner.py:657]: **todo:** Remove; this belongs in the other PR.

File: docs/file_formats.md [hunk 5]
```diff
@@ -9,8 +9,9 @@ When you are ready to use Casanovo, you can input your MS/MS spectra in one of t
 - **[mzML](https://doi.org/10.1074/mcp.R110.000133)**: XML-based mass spectrometry community standard file format developed by the Proteomics Standards Initiative (PSI).
 - **[mzXML](https://doi.org/10.1038/nbt1031)**: XML-based predecessor of mzML. Although supported by Casanovo, mzML should typically be preferred instead.
 - **[MGF](https://www.matrixscience.com/help/data_file_help.html)**: A simple text-based peak file, though not as rich in detail as mzML.
+- **Bruker '.d' folder**: A proprietary directory-based raw data format used by Bruker instruments.
```
Comment (bittremieux) [inline docs/file_formats.md:12]: **todo:** Remove; belongs in separate PR.

File: tests/unit_tests/test_runner.py [hunk 6]
```diff
@@ -14,6 +14,36 @@
 from casanovo.denovo.model_runner import ModelRunner
 
 
+def test_loading_timstof_folders(tmp_path, monkeypatch):
```
Comment (bittremieux) [inline tests/unit_tests/test_runner.py:17]: **todo:** Remove.

File: tests/unit_tests/test_unit.py [hunk 7]
```diff
@@ -998,18 +775,18 @@ def test_length_restrictions(tiny_fasta_file):
...
+            "nterm:+42.011,nterm:+43.006,nterm:-17.027,nterm:+43.006-17.027"
         ),
         tokenizer=depthcharge.tokenizers.PeptideTokenizer.from_massivekb(),
     )
     assert pdb.db_peptides.index.to_list() == expected_short
 
 
 @pytest.mark.parametrize(
-    "enzyme,digestion,expected",
+    ("enzyme,digestion,expected"),
```
Comment (bittremieux) [inline tests/unit_tests/test_unit.py:789]: **todo:** I don't think these parentheses do anything, you're still missing a trailing comma to make it a tuple.

File: tests/test_integration.py [hunk 8]
```diff
@@ -80,7 +80,7 @@ def test_train_and_run(
     # Verify that the spectrum predictions are correct and indexed
     # according to the peak input file type.
     psms = mztab.spectrum_match_table
-    assert psms.loc[1, "sequence"] == "LESLLEK"
+    assert psms.loc[1, "sequence"] == "LESLEK"
```
Comment (bittremieux) [inline tests/test_integration.py:83]: **question:** How come these peptides have been changed from the previous test version?

File: CHANGELOG.md [hunk 9]
```diff
@@ -8,22 +8,17 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ### Added
 
-- A TSV file with all candidate peptides can be exported during database searching with the `--export` flag.
+- timsTOF files (in the format of .d folders) can now be accepted as spectra input files.
```
Comment (bittremieux) [inline CHANGELOG.md:11]: **todo:** Remove.

**Review by bittremieux — CHANGES_REQUESTED** [review]
It was quite challenging to review this PR because it mixes various different things: evaluation metrics (although no actual functional changes, only type hints), gradient clipping, checkpoint vs config loading, timsTOF changes, and some more random small things.

Best practice is to have one change per PR. This makes it much easier to implement and review a coherent change. Please check the comments and then split the different contributions where still relevant into separate PRs; meanwhile I'm going to close this one.

File: casanovo/denovo/model_runner.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (bittremieux) [reply casanovo/denovo/model_runner.py:529]: Yes, I know. So the original code was correct, and swapping the two parameters in this PR seems incorrect, which is what my comment indicates.
